tools/yulPhaser/SimulationRNG.cpp

```diff
@@ -35,15 +35,15 @@ bool SimulationRNG::bernoulliTrial(double _successProbability)
 	return static_cast<bool>(distribution(s_generator));
 }
 
-uint32_t SimulationRNG::uniformInt(uint32_t _min, uint32_t _max)
+long SimulationRNG::uniformInt(long _min, long _max)
```

Comment:
```suggestion
size_t SimulationRNG::uniformInt(size_t _min, size_t _max)
```

Why not keep the parameters and result unsigned? You wouldn't have to insert those `long` conversions all over the code.

Reply:
You are right. There was one reason I did change types... Binomial distribution doesn't accept signed type as a template parameter as it doesn't compile (can't resolve `abs()` of signed type...). If I leave it with signed type the only usage of function parameters was inside of cast and compiler produce an error for that... I'll try once more